lib/compile.js

```diff
@@ -105,6 +106,26 @@ function compileLoad(loader, load, compileOpts, cache) {
     if (format == 'cjs')
       mappedLoad.source = mappedLoad.source.replace(hashBangRegEx, '');
     return Promise.resolve()
+    .then(function () {
```

Comment:
The only issue with this approach is if the fetched source is not JavaScript, but I think the source map detection can work equally across input types fine so this won't be a bad failure, worst case we return `metadata.sourceMap` when we shouldn't. But we should ensure this process itself is resilient to errors at least.